gdx/src/com/badlogic/gdx/graphics/g2d/Sprite.java

```diff
@@ -338,45 +339,50 @@ public void translate (float xAmount, float yAmount) {
 	/** Sets the color used to tint this sprite. Default is {@link Color#WHITE}. */
 	public void setColor (Color tint) {
 		color.set(tint);
-		float color = tint.toFloatBits();
+		packedColor = tint.toFloatBits();
 		float[] vertices = this.vertices;
-		vertices[C1] = color;
-		vertices[C2] = color;
-		vertices[C3] = color;
-		vertices[C4] = color;
+		vertices[C1] = packedColor;
+		vertices[C2] = packedColor;
+		vertices[C3] = packedColor;
+		vertices[C4] = packedColor;
 	}
 
 	/** Sets the alpha portion of the color used to tint this sprite. */
 	public void setAlpha (float a) {
-		color.a = a;
-		float color = this.color.toFloatBits();
-		vertices[C1] = color;
-		vertices[C2] = color;
-		vertices[C3] = color;
-		vertices[C4] = color;
+		if (color.a != a) {
+			color.a = a;
+			packedColor = color.toFloatBits();
+			vertices[C1] = packedColor;
+			vertices[C2] = packedColor;
+			vertices[C3] = packedColor;
+			vertices[C4] = packedColor;
+		}
 	}
 
 	/** @see #setColor(Color) */
 	public void setColor (float r, float g, float b, float a) {
 		color.set(r, g, b, a);
-		float color = this.color.toFloatBits();
+		packedColor = color.toFloatBits();
 		float[] vertices = this.vertices;
-		vertices[C1] = color;
-		vertices[C2] = color;
-		vertices[C3] = color;
-		vertices[C4] = color;
+		vertices[C1] = packedColor;
+		vertices[C2] = packedColor;
+		vertices[C3] = packedColor;
+		vertices[C4] = packedColor;
 	}
 
 	/** Sets the color of this sprite, expanding the alpha from 0-254 to 0-255.
```

Comment:
It's not an issue if the == comparison is different and they pack the same float, just misses the optmization on these corner cases. Problem is when the opposite happens.

Reply:
It's not an issue if the == comparison is different and they pack the same float, just misses the optmization on these corner cases. Problem is when the opposite happens.

Reply:
I guess it means setting a 0xFE alpha packed float color will result in a 1f alpha `Color` which will be rendered as fully opaque instead of the "expected" slight transparency. Even if that was the case I agree we should remove it as it may cause more confusion than anything. I'll make the change.